analyses/NHRC/olivia-refactor/preprocess_and_save.py

```diff
@@ -0,0 +1,241 @@
+        gap_start = accel_data[gap_index, 0] + ACC_DIFF_GAP
+        gap_end = accel_data[gap_index + 1, 0]
+        mask_indices = np.where(
+            (psg_data[:, 0] + PSG_DT >= gap_start) &
+            (psg_data[:, 0] <= gap_end))[0]
+        wake_counts = np.sum((psg_data[mask_indices, 1].astype(int)) == 0)
+        wakes_masked += wake_counts
+        psg_data[mask_indices, 1:] = -1
+
+    post_mask_sleeps = np.sum(psg_data[:, 1] > 0)
+    post_mask_wakes = np.sum(psg_data[:, 1] == 0)
+    print("Post-mask:\n\tSleeps", post_mask_sleeps, "\n\tWakes",
+          post_mask_wakes, "\n\tWakes masked", wakes_masked)
+
+    # Convert accelerometer data to spectrograms
+
+    sample_rate = 50
+    if acc_Hz_str == "dyn":
+        int_hz = int(avg_time_hz)
+        print("dynamic rate:", int_hz)
+        sample_rate = int_hz
+    else:
+        sample_rate = int(acc_Hz_str)
+        print("fixed rate:", sample_rate)
+    accel_data_resampled = resample_accel_data(
+        accel_data, original_fs=sample_rate, target_fs=ACC_INPUT_HZ)
+
+    # compute spectrogram with resampled data
+    spectrograms = processor.accelerometer_to_spectrogram(
+        accel_data_resampled)
+    padded_spectrograms = np.zeros(FIXED_SPECGRAM_SHAPE)
+    padded_spectrograms[:spectrograms.shape[0],
+                        ...] = spectrograms[:FIXED_SPECGRAM_SHAPE[0], ...]
+
+    # Compute activity with resampled data
+    activity_time, ads = build_ADS(accel_data)
+    activity_data = np.column_stack((activity_time, ads))
+
+    # Pad PSG data to 1024 samples
+    psg_data = pad_or_truncate(psg_data, int(FIXED_LABEL_LENGTH))
+
+    # Pad activity data to 2 * 1024 samples
+    activity_data = pad_or_truncate(activity_data, int(LR_ACTIVITY_INPUTS))
+
+    return {"spectrogram": padded_spectrograms,
+            "activity": activity_data,
+            "psg": psg_data}
+
+
+def process_data_set(data_set: pds.DataSetObject,
+                     ids_to_exclude: List[str],
+                     processor) -> Dict[str, Dict[str, np.ndarray]]:
+    data = {}
+    for subject_id in data_set.ids:
+        if subject_id in ids_to_exclude:
+            continue
+        print(f"Processing {subject_id}")
+        data[subject_id] = process_data(
+            data_set,
+            processor,
+            subject_id)
+    return data
+
+
+def pad_or_truncate(data,
+                    desired_length,
+                    pad_value: float = -1.0):
+    current_length = data.shape[0]
+    if current_length < desired_length:
+        padding = desired_length - current_length
+        pad_width = ((0, padding), (0, 0))
+        data = np.pad(data, pad_width, mode='constant',
+                      constant_values=pad_value)
+    else:
+        data = data[:desired_length, :]
+    return data
+
+
+def do_preprocessing():
+    # clean_and_save_accelerometer_data()
+
+    start_run = time.time()
+    data_location = Path('/Users/ojwalch/Documents/eric-pisces/datasets')
```

Comment:
not opposed to the hardcoded path, but I see `Path('/Users/ojwalch/Documents/eric-pisces/datasets')` (and children of that dir) repeated. Let's pull that into a variable and then `DATA_PATH.joinpath(...)`